### PR TITLE
Fix flapping acceptance test

### PR DESCRIPTION
**Pull request by QuentinBisson**
This PR fixes the flapping TestRetry acceptance test:

```bash
for i in {1..20}
do
  go test . -run TestRetry -count 1
done
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.514s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.515s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.513s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.514s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.512s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.513s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.514s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.515s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.512s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.514s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.511s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.513s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.514s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.514s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.513s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.514s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.513s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.515s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.514s
ok      github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.515s
```

### CODE REVIEW

**Comment by QuentinBisson** [comment]
cc @roidelapluie as we talked about it in https://github.com/prometheus/alertmanager/issues/1682

I am not sure why the checks are not reported though

**Comment by roidelapluie** [comment]
Is it only flapping in go 1.16 or did we miss it?

**Comment by QuentinBisson** [comment]
I reproduced it with go 1.15.11:

```bash
GOROOT=/home/quentin/Downloads/go1.15.11.linux-amd64/go
> for i in {1..20}
do
  go test . -run TestRetry -count 1
done
--- FAIL: TestRetry (6.50s)
    acceptance.go:179: stdout:
    acceptance.go:180: stderr:
        level=info ts=2021-04-27T09:16:29.863Z caller=main.go:221 msg="Starting Alertmanager" version="(version=0.21.0, branch=master, revision=87915dc6521ac5344662d46b910b9e6b4ff5f355)"
        level=info ts=2021-04-27T09:16:29.863Z caller=main.go:222 build_context="(go=go1.16.3, user=quentin@quentin-gs, date=20210423-12:23:58)"
        level=debug ts=2021-04-27T09:16:29.863Z caller=cluster.go:172 component=cluster msg="resolved peers to following addresses" peers=127.0.0.1:40267
        level=debug ts=2021-04-27T09:16:29.863Z caller=delegate.go:235 component=cluster received=NotifyJoin node=01F499V797XYE1JAXKV8HAA1MW addr=127.0.0.1:40267
        level=debug ts=2021-04-27T09:16:29.864Z caller=cluster.go:329 component=cluster memberlist="2021/04/27 11:16:29 [DEBUG] memberlist: Initiating push/pull sync with:  127.0.0.1:40267\n"
        level=debug ts=2021-04-27T09:16:29.864Z caller=cluster.go:329 component=cluster memberlist="2021/04/27 11:16:29 [DEBUG] memberlist: Stream connection from=127.0.0.1:56314\n"
        level=debug ts=2021-04-27T09:16:29.864Z caller=cluster.go:256 component=cluster msg="joined cluster" peers=1
        level=info ts=2021-04-27T09:16:29.864Z caller=cluster.go:671 component=cluster msg="Waiting for gossip to settle..." interval=2s
        level=info ts=2021-04-27T09:16:29.864Z caller=cluster.go:680 component=cluster msg="gossip not settled but continuing anyway" polls=0 elapsed=1.33µs
        level=debug ts=2021-04-27T09:16:29.891Z caller=main.go:360 externalURL=http://quentin-gs:33163
        level=info ts=2021-04-27T09:16:29.891Z caller=coordinator.go:113 component=configuration msg="Loading configuration file" file=/tmp/am_test071431835/config.yml
        level=info ts=2021-04-27T09:16:29.891Z caller=coordinator.go:126 component=configuration msg="Completed loading of configuration file" file=/tmp/am_test071431835/config.yml
        level=debug ts=2021-04-27T09:16:29.893Z caller=main.go:477 routePrefix=/
        level=info ts=2021-04-27T09:16:29.894Z caller=main.go:497 msg=Listening address=127.0.0.1:33163
        level=info ts=2021-04-27T09:16:29.894Z caller=tls_config.go:191 msg="TLS is disabled." http2=false
        level=debug ts=2021-04-27T09:16:30.851Z caller=dispatch.go:138 component=dispatcher msg="Received alert" alert=test1[5ae90ff][active]
        level=debug ts=2021-04-27T09:16:31.852Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
        level=debug ts=2021-04-27T09:16:31.852Z caller=notify.go:734 component=dispatcher receiver=default integration=webhook[1] msg="Notify success" attempts=1
        level=debug ts=2021-04-27T09:16:31.853Z caller=notify.go:734 component=dispatcher receiver=default integration=webhook[0] msg="Notify success" attempts=1
        level=debug ts=2021-04-27T09:16:32.853Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
        level=debug ts=2021-04-27T09:16:33.853Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
        level=debug ts=2021-04-27T09:16:34.854Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
        level=debug ts=2021-04-27T09:16:34.855Z caller=notify.go:734 component=dispatcher receiver=default integration=webhook[1] msg="Notify success" attempts=1
        level=debug ts=2021-04-27T09:16:34.855Z caller=notify.go:734 component=dispatcher receiver=default integration=webhook[0] msg="Notify success" attempts=1
        level=debug ts=2021-04-27T09:16:35.854Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
    send_test.go:217: 
        collector "webhook":
        
        interval [2,2.5]
        ---
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:30.849+02:00 <nil> <nil> { map[alertname:test1]}}[1:]
          [ ✓ ]
        interval [6,6.5]
        ---
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:30.849+02:00 <nil> <nil> { map[alertname:test1]}}[1:]
          [ ✗ ]
        
        received:
        @ 2.003446067
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:30.851+02:00 <nil> <nil> { map[alertname:test1]}}[1.00195564:]
        @ 5.00571752
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:30.851+02:00 <nil> <nil> { map[alertname:test1]}}[1.00195564:]
        
    send_test.go:217: 
        collector "webhook_failing":
        
        interval [6,6.5]
        ---
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:30.849+02:00 <nil> <nil> { map[alertname:test1]}}[1:]
          [ ✗ ]
        
        received:
        @ 5.005628204
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:30.851+02:00 <nil> <nil> { map[alertname:test1]}}[1.00195564:]
        
FAIL
FAIL    github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.516s
FAIL
^C--- FAIL: TestRetry (6.50s)
    acceptance.go:179: stdout:
    acceptance.go:180: stderr:
        level=info ts=2021-04-27T09:16:41.357Z caller=main.go:221 msg="Starting Alertmanager" version="(version=0.21.0, branch=master, revision=87915dc6521ac5344662d46b910b9e6b4ff5f355)"
        level=info ts=2021-04-27T09:16:41.357Z caller=main.go:222 build_context="(go=go1.16.3, user=quentin@quentin-gs, date=20210423-12:23:58)"
        level=debug ts=2021-04-27T09:16:41.357Z caller=cluster.go:172 component=cluster msg="resolved peers to following addresses" peers=127.0.0.1:33559
        level=debug ts=2021-04-27T09:16:41.357Z caller=delegate.go:235 component=cluster received=NotifyJoin node=01F499VJGDTPNB7N90VJG13GZR addr=127.0.0.1:33559
        level=debug ts=2021-04-27T09:16:41.358Z caller=cluster.go:329 component=cluster memberlist="2021/04/27 11:16:41 [DEBUG] memberlist: Initiating push/pull sync with:  127.0.0.1:33559\n"
        level=debug ts=2021-04-27T09:16:41.358Z caller=cluster.go:329 component=cluster memberlist="2021/04/27 11:16:41 [DEBUG] memberlist: Stream connection from=127.0.0.1:35280\n"
        level=debug ts=2021-04-27T09:16:41.358Z caller=cluster.go:256 component=cluster msg="joined cluster" peers=1
        level=info ts=2021-04-27T09:16:41.358Z caller=cluster.go:671 component=cluster msg="Waiting for gossip to settle..." interval=2s
        level=info ts=2021-04-27T09:16:41.358Z caller=cluster.go:680 component=cluster msg="gossip not settled but continuing anyway" polls=0 elapsed=2µs
        level=debug ts=2021-04-27T09:16:41.382Z caller=main.go:360 externalURL=http://quentin-gs:39877
        level=info ts=2021-04-27T09:16:41.382Z caller=coordinator.go:113 component=configuration msg="Loading configuration file" file=/tmp/am_test983764908/config.yml
        level=info ts=2021-04-27T09:16:41.382Z caller=coordinator.go:126 component=configuration msg="Completed loading of configuration file" file=/tmp/am_test983764908/config.yml
        level=debug ts=2021-04-27T09:16:41.384Z caller=main.go:477 routePrefix=/
        level=info ts=2021-04-27T09:16:41.384Z caller=main.go:497 msg=Listening address=127.0.0.1:39877
        level=info ts=2021-04-27T09:16:41.384Z caller=tls_config.go:191 msg="TLS is disabled." http2=false
        level=debug ts=2021-04-27T09:16:42.349Z caller=dispatch.go:138 component=dispatcher msg="Received alert" alert=test1[5ae90ff][active]
        level=debug ts=2021-04-27T09:16:43.351Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
        level=debug ts=2021-04-27T09:16:43.353Z caller=notify.go:734 component=dispatcher receiver=default integration=webhook[1] msg="Notify success" attempts=1
        level=debug ts=2021-04-27T09:16:43.353Z caller=notify.go:734 component=dispatcher receiver=default integration=webhook[0] msg="Notify success" attempts=1
        level=debug ts=2021-04-27T09:16:44.352Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
        level=debug ts=2021-04-27T09:16:45.353Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
        level=debug ts=2021-04-27T09:16:46.353Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
        level=debug ts=2021-04-27T09:16:46.354Z caller=notify.go:734 component=dispatcher receiver=default integration=webhook[1] msg="Notify success" attempts=1
        level=debug ts=2021-04-27T09:16:46.354Z caller=notify.go:734 component=dispatcher receiver=default integration=webhook[0] msg="Notify success" attempts=1
        level=debug ts=2021-04-27T09:16:47.354Z caller=dispatch.go:474 component=dispatcher aggrGroup="{}:{alertname=\"test1\"}" msg=flushing alerts=[test1[5ae90ff][active]]
    send_test.go:217: 
        collector "webhook":
        
        interval [2,2.5]
        ---
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:42.347+02:00 <nil> <nil> { map[alertname:test1]}}[1:]
          [ ✓ ]
        interval [6,6.5]
        ---
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:42.347+02:00 <nil> <nil> { map[alertname:test1]}}[1:]
          [ ✗ ]
        
        received:
        @ 2.005587303
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:42.349+02:00 <nil> <nil> { map[alertname:test1]}}[1.002038012:]
        @ 5.006824458
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:42.349+02:00 <nil> <nil> { map[alertname:test1]}}[1.002038012:]
        
    send_test.go:217: 
        collector "webhook_failing":
        
        interval [6,6.5]
        ---
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:42.347+02:00 <nil> <nil> { map[alertname:test1]}}[1:]
          [ ✗ ]
        
        received:
        @ 5.006861775
        - &{map[] 0001-01-01T00:00:00.000Z <nil> [] 2021-04-27T11:16:42.349+02:00 <nil> <nil> { map[alertname:test1]}}[1.002038012:]
        
FAIL
FAIL    github.com/prometheus/alertmanager/test/with_api_v2/acceptance  6.515s
```

**Comment by QuentinBisson** [comment]
I cannot reproduce it with go 1.15.11. I tested the code with both `group_interval:  1s` and `group_interval:  2s` and the test are running successfully in both cases

**Comment by roidelapluie** [comment]
Can you please also update the .circleci.yml  to use go 1.16 for the acceptance tests?

Thanks!!

**Comment by QuentinBisson** [comment]
This is causing other tests to fail now :(

**Comment by roidelapluie** [comment]
I have relaunched the tests and they fail with another error..

**Comment by QuentinBisson** [comment]
After some trials, I see the text passing but the build is failing because go.sum is changing after the make proto step

**Comment by roidelapluie** [comment]
https://github.com/prometheus/prometheus/commit/75e505babb9bbcefd8849e945814d35c7ce97e9f#diff-beb85cf1fc0ad937994d001f06b66b2a0e87a9416cc722aa97849b0d6ba17cb0

I copy back and forth the go.sum in prometheus. We could do the same.

**Comment by QuentinBisson** [comment]
Oh interesting :)

**Comment by QuentinBisson** [comment]
This seems to have fixed it :D

File: scripts/genproto.sh [hunk 1]
```diff
@@ -46,3 +51,5 @@ for dir in ${DIRS}; do
         goimports -w *.pb.go
     popd
 done
+
+mv go.sum.bak go.sum
```
Comment (beorn7) [inline scripts/genproto.sh:55]: EOL missing.

**Review by beorn7 — COMMENTED** [review]
Just a tiny nit.

I'm missing the context to say if the `go.sum` hack is kosher and if the increased group interval makes sense. But in doubt, I'd say let's merge this and release. We can always improve/revert/…

File: scripts/genproto.sh (hunk 1 above)
Comment (QuentinBisson) [reply scripts/genproto.sh]: Solved

**Review by beorn7 — APPROVED** [review]

**Comment by beorn7** [comment]
Thanks @QuentinBisson .

@roidelapluie @simonpasquier I'm merging this now, to unblock the much needed release. I won't have time to shepherd it before end of next week. Will any of you be able to?

**Pull request merged**